### PR TITLE
Bump ci-kubernetes-verify-* memory to 24 Gi

### PR DESCRIPTION
**Pull request by hasheddan**
Verify jobs are routinely hitting OOM exceptions on typecheck. This
gives them a little more room but not so much that large outliers will
not surface as failures.

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

Part 3 of #18508 

/cc @spiffxp @BenTheElder

This will get these jobs running again but I want to look into the infrequent spikes where nearly double the amount of memory is used. Helpful query: https://console.cloud.google.com/monitoring/metrics-explorer?project=k8s-infra-prow-build&timeDomain=6h&pageState=%7B%22xyChart%22:%7B%22dataSets%22:%5B%7B%22timeSeriesFilter%22:%7B%22filter%22:%22metric.type%3D%5C%22kubernetes.io%2Fcontainer%2Fmemory%2Fused_bytes%5C%22%20resource.type%3D%5C%22k8s_container%5C%22%20metadata.user_labels.%5C%22prow.k8s.io%2Fjob%5C%22%3Dmonitoring.regex.full_match(%5C%22ci-kubernetes-verify-.*%5C%22)%22,%22minAlignmentPeriod%22:%2260s%22,%22unitOverride%22:%22By%22,%22aggregations%22:%5B%7B%22perSeriesAligner%22:%22ALIGN_MEAN%22,%22crossSeriesReducer%22:%22REDUCE_SUM%22,%22groupByFields%22:%5B%22metadata.user_labels.%5C%22prow.k8s.io%2Fjob%5C%22%22%5D%7D,%7B%22crossSeriesReducer%22:%22REDUCE_NONE%22%7D%5D%7D,%22targetAxis%22:%22Y1%22,%22plotType%22:%22LINE%22%7D%5D,%22options%22:%7B%22mode%22:%22COLOR%22%7D,%22constantLines%22:%5B%5D,%22timeshiftDuration%22:%220s%22,%22y1Axis%22:%7B%22label%22:%22y1Axis%22,%22scale%22:%22LINEAR%22%7D%7D,%22isAutoRefresh%22:true,%22timeSelection%22:%7B%22timeRange%22:%226h%22%7D%7D

### CODE REVIEW

**Comment by spiffxp** [comment]
```
    jobs_test.go:999: ci-kubernetes-verify-stable3: must have non-zero resource.requests == resource.limits to run on k8s-infra-prow-build cluster, has <snip>
```
tl;dr some part of the limit isn't matching the request.  Default formatting for resource quantities is ugly.

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.16.yaml [hunk 1]
```diff
@@ -436,7 +436,7 @@ periodics:
       resources:
         limits:
           cpu: "6"
-          memory: 16Gi
+          memory: 24Gi
         requests:
           cpu: "6"
           memory: 16Gi
```
Comment (spiffxp) [inline config/jobs/kubernetes/sig-release/release-branch-jobs/1.16.yaml:442]: request.memory needs to be bumped as well

Comment (hasheddan) [reply config/jobs/kubernetes/sig-release/release-branch-jobs/1.16.yaml:442]: Ah missed that one, thanks!

**Comment by BenTheElder** [comment]
this seems like a lot

**Comment by hasheddan** [comment]
@BenTheElder I agree, but recent runs suggest it still isn't even enough 😕 Need to figure out what's up with typecheck
